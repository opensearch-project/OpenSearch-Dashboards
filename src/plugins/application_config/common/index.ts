export const PLUGIN_ID = 'applicationConfig';
export const PLUGIN_NAME = 'applicationConfig';
