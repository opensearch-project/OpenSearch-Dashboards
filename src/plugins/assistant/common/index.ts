/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const PLUGIN_ID = 'assistant';
export const PLUGIN_NAME = 'assistant';

export * from './types';
