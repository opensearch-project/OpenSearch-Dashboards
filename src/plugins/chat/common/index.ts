/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const PLUGIN_ID = 'chat';
export const PLUGIN_NAME = 'chat';
export const CHAT_DEFAULT_AG_UI_URL = 'http://localhost:3000';
