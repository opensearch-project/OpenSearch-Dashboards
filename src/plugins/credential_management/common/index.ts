/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const PLUGIN_ID = 'credentialManagement';
export const PLUGIN_NAME = 'credentialManagement';
