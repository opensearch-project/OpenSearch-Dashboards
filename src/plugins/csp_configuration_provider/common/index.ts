/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const PLUGIN_ID = 'cspConfigurationProvider';
export const PLUGIN_NAME = 'CspConfigurationProvider';
