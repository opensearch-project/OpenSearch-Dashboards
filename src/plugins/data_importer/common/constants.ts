/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const PLUGIN_NAME_AS_TITLE = 'Data Importer';
export const PLUGIN_ID = 'dataImporter';
export const PLUGIN_NAME = 'dataImporter';

export const CSV_FILE_TYPE = 'csv';
export const JSON_FILE_TYPE = 'json';
export const NDJSON_FILE_TYPE = 'ndjson';

export const DEFAULT_SUPPORTED_FILE_TYPES_LIST = [CSV_FILE_TYPE, JSON_FILE_TYPE, NDJSON_FILE_TYPE];

export const CSV_SUPPORTED_DELIMITERS = [',', ';', '\t', '|'];

export enum DYNAMIC_MAPPING_TYPES {
  NULL = 'null',
  BOOLEAN = 'boolean',
  FLOAT = 'float',
  DOUBLE = 'double',
  INTEGER = 'integer',
  OBJECT = 'object',
  ARRAY = 'array',
  TEXT = 'text',
  KEYWORD = 'keyword',
  DATE = 'date',
}
