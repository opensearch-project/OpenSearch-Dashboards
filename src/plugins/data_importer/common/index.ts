export const PLUGIN_ID = 'dataImporter';
export const PLUGIN_NAME = 'Data Importer';
