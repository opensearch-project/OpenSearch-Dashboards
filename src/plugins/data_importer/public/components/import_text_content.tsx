/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useState } from 'react';

import {
  EuiCodeEditor,
  EuiFlexGroup,
  EuiFlexItem,
  EuiPageContentBody,
  EuiSelect,
  EuiText,
} from '@elastic/eui';
import uuid from 'uuid';
import { JSON_FILE_TYPE } from '../../common/constants';

export interface ImportTextContentBodyProps {
  onTextChange: (text: string) => void;
  enabledFileTypes: string[];
  onFileTypeChange: (fileType: string) => void;
  characterLimit: number;
  initialFileType: string;
}

export const ImportTextContentBody = ({
  onTextChange,
  enabledFileTypes,
  onFileTypeChange,
  characterLimit,
  initialFileType,
}: ImportTextContentBodyProps) => {
  const [codeEditorText, setCodeEditorText] = useState<string>('');
  const [fileType, setFileType] = useState<string>(initialFileType);
  const [numCharacters, setNumCharacters] = useState<number>(0);
  const options = enabledFileTypes.map((type) => {
    return {
      value: type,
      text: type,
    };
  });

  const onTextUpdate = (text: string) => {
    setCodeEditorText(text);
    onTextChange(text);
    setNumCharacters(text.length);
  };

  const onOptionSelect = (e: any) => {
    setFileType(e.target.value === JSON_FILE_TYPE ? e.target.value : undefined);
    onFileTypeChange(e.target.value);
  };

  return (
    <EuiPageContentBody>
      <EuiFlexGroup direction="column">
        <EuiFlexItem>
          <EuiFlexGroup justifyContent="flexStart" gutterSize="s">
            <EuiFlexItem grow={false}>
              <EuiText>File Format: </EuiText>
            </EuiFlexItem>
            <EuiFlexItem grow={false}>
              <EuiSelect
                options={options}
                fullWidth={false}
                onChange={onOptionSelect}
                value={initialFileType}
              />
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiCodeEditor
            id={uuid.v4()}
            onChange={onTextUpdate}
            width={'full'}
            value={codeEditorText}
            mode={fileType}
          />
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiText color={numCharacters > characterLimit ? 'danger' : 'default'}>
            {numCharacters}/{characterLimit} characters
          </EuiText>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiPageContentBody>
  );
};
