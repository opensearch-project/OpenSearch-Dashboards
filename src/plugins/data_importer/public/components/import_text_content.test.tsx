/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from '@testing-library/react';
import { ImportTextContentBody } from './import_text_content';
import { DEFAULT_SUPPORTED_FILE_TYPES_LIST } from '../../common';

jest.mock('uuid', () => ({
  v4: jest.fn(() => 'mocked-id'),
}));

describe('ImportTextContent', () => {
  it('should render', () => {
    const { container } = render(
      <ImportTextContentBody
        onTextChange={jest.fn()}
        enabledFileTypes={DEFAULT_SUPPORTED_FILE_TYPES_LIST}
        onFileTypeChange={jest.fn()}
        characterLimit={10000}
        initialFileType={DEFAULT_SUPPORTED_FILE_TYPES_LIST[0]}
      />
    );

    // These id fields cannot be manually set
    container.querySelector('div.ace_editor')?.removeAttribute('id');
    container.querySelector('button.euiCodeEditorKeyboardHint')?.removeAttribute('id');

    expect(container).toMatchSnapshot();
  });
});
