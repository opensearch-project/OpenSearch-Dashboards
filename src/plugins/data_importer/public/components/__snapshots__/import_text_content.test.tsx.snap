// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportTextContent should render 1`] = `
<div>
  <div
    class="euiPageContentBody"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionColumn euiFlexGroup--responsive"
    >
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <div
              class="euiText euiText--medium"
            >
              File Format: 
            </div>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <div
              class="euiFormControlLayout"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <select
                  class="euiSelect"
                >
                  <option
                    value="csv"
                  >
                    csv
                  </option>
                  <option
                    value="json"
                  >
                    json
                  </option>
                  <option
                    value="ndjson"
                  >
                    ndjson
                  </option>
                </select>
                <div
                  class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                >
                  <span
                    class="euiFormControlLayoutCustomIcon"
                  >
                    <span
                      aria-hidden="true"
                      class="euiFormControlLayoutCustomIcon__icon"
                      data-euiicon-type="arrowDown"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem"
      >
        <div
          class="euiCodeEditorWrapper"
          data-test-subj="codeEditorContainer"
        >
          <button
            class="euiCodeEditorKeyboardHint"
            data-test-subj="codeEditorHint"
          >
            <p
              class="euiText"
            >
              Press Enter to start editing.
            </p>
            <p
              class="euiText"
            >
              When you're done, press Escape to stop editing.
            </p>
          </button>
          <div
            class=" ace_editor ace-tm"
            style="height: 500px;"
          >
            <textarea
              autocapitalize="off"
              autocorrect="off"
              class="ace_text-input"
              id="mocked-id"
              spellcheck="false"
              style="opacity: 0;"
              tabindex="-1"
              wrap="off"
            />
            <div
              aria-hidden="true"
              class="ace_gutter"
            >
              <div
                class="ace_layer ace_gutter-layer ace_folding-enabled"
              />
              <div
                class="ace_gutter-active-line"
              />
            </div>
            <div
              class="ace_scroller"
            >
              <div
                class="ace_content"
              >
                <div
                  class="ace_layer ace_print-margin-layer"
                >
                  <div
                    class="ace_print-margin"
                    style="left: 4px; visibility: visible;"
                  />
                </div>
                <div
                  class="ace_layer ace_marker-layer"
                />
                <div
                  class="ace_layer ace_text-layer"
                  style="padding: 0px 4px;"
                />
                <div
                  class="ace_layer ace_marker-layer"
                />
                <div
                  class="ace_layer ace_cursor-layer ace_hidden-cursors"
                >
                  <div
                    class="ace_cursor"
                  />
                </div>
              </div>
            </div>
            <div
              class="ace_scrollbar ace_scrollbar-v"
              style="display: none; width: 20px;"
            >
              <div
                class="ace_scrollbar-inner"
                style="width: 20px;"
              />
            </div>
            <div
              class="ace_scrollbar ace_scrollbar-h"
              style="display: none; height: 20px;"
            >
              <div
                class="ace_scrollbar-inner"
                style="height: 20px;"
              />
            </div>
            <div
              style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: hidden;"
            >
              <div
                style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
              />
              <div
                style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
              >
                XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem"
      >
        <div
          class="euiText euiText--medium"
        >
          <div
            class="euiTextColor euiTextColor--default"
          >
            0
            /
            10000
             characters
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
